cmake/kokkos_tpls.cmake

```diff
@@ -13,7 +13,12 @@ KOKKOS_TPL_OPTION(LIBNUMA Off)
 KOKKOS_TPL_OPTION(MEMKIND Off)
 KOKKOS_TPL_OPTION(CUDA    Off)
 KOKKOS_TPL_OPTION(LIBRT   Off)
-KOKKOS_TPL_OPTION(LIBDL   On)
+MESSAGE(STATUS "COMPILER ID ${CMAKE_CXX_COMPILER_ID}")
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
```

Comment:
how about `if(WIN32)`? That would also catch the "we use nvcc as the compiler on windows" case